InAppUtils/InAppUtils.m

```diff
@@ -116,6 +127,27 @@ - (void) doPurchaseProduct:(NSString *)productIdentifier
         if(username) {
             payment.applicationUsername = username;
         }
+
+        if (discountSignature && discountIdentifier && username) {
```

Comment:
could unwrap this if and make it if (discountSignature && discountIdentifier) then check for username and return if !username up top-- also, is there an error callback that should be called if !username? but either way.

Reply:
Thank you. Yes, great idea. If there isn't a username when purchasing a discount product it will fail. (I'll add the callback for that).  Currently for regular purchases we haven't been adding that applicationUsername to the payment, so we could keep that the same?  - Just have to wait for a sandbox sub to expire so I can test again!

Reply:
ok - added callback for invalid username when making discount purchase.